cli/runtime/config/nodeutils/replace_nodes_test.go

```diff
@@ -0,0 +1,280 @@
+// Copyright 2022 VMware, Inc. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package nodeutils
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"gopkg.in/yaml.v3"
+)
+
+func TestReplaceNodes(t *testing.T) {
+	tests := []struct {
+		name     string
+		dst      string
+		src      string
+		metadata map[string]string
+		output   string
+	}{
+		{
+			name:     "success replace nodes with empty patch strategy",
+			metadata: make(map[string]string),
+			dst: `name: test-mc
+type: k8s
```

Comment:
nit: the test data is unnecessarily complex, containing too much stuff that wasn't germane to demonstrating the node replacing (actual deleting) effect. It is difficult to manually diff 3 sets of yaml content when one cannot keep most of the data on screen simultaneously. I suggest we pare it down some without sacrificing coverage.

Reply:
refactored tests